src/internal/useObservable.ts

```diff
@@ -51,5 +52,8 @@ export const useObservable = <O>(
     return () => subscription.unsubscribe()
   }, [source$])
 
-  return state !== (SUSPENSE as any) ? (state as any) : source$.getValue()
+  if (state === SUSPENSE) {
+    throw source$.getValue()
+  }
+  return state
```

Comment:
Can't this have some sort of race condition where `source$.getValue()` would return a value when `state === SUSPENSE`?

Reply:
Mmmmm _maybe_, as in we dispatch `SUSPENSE` from the `useEffect`, but since it's async, then by the time that we return, then the promise is gone... yeah, that's a problem... Thanks for pointing that out!

Reply:
Done!